### PR TITLE
Add tofuNFT into nft.trades on V2

### PR DESCRIPTION
**Pull request by theachenyj**
Brief comments on the purpose of your changes:
Adding tofu's direct marketplace into Dune v2
**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by Hosuke** [comment]
Hi @theachenyj @0xRobin , I just merged the main into this PR based on [#PR2315](https://github.com/duneanalytics/spellbook/pull/2315) mentioned. Please have a look if everything still works in this PR.

**Comment by theachenyj** [comment]
> 

Get it. I will check it asap.

**Comment by theachenyj** [comment]
> Hi @theachenyj @0xRobin , I just merged the main into this PR based on [#PR2315](https://github.com/duneanalytics/spellbook/pull/2315) mentioned. Please have a look if everything still works in this PR.

I'v check and fix some bugs. It looks good now.

**Review by 0xRobin — APPROVED** [review]
I added some query tweaks and a test with a seed file. 
Checks are passing, this is ready to be merged on the 28th.
Thanks for the contribution!

File: models/tofu/bnb/tofu_bnb_events.sql [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+     , tfe.evt_block_time                    as block_time
+     , tfe.evt_block_number                  as block_number
+     , tff.token_id                          as token_id
+     , nft.standard                          as token_standard
+     , nft.name                              as collection
+     , case
+           when tff.bundle_size = 1 then 'Single Item Trade'
+           else 'Bundle Trade'
+    end                                      as trade_type
+     , tff.amount                            as number_of_items
```
Comment (Hosuke) [inline models/tofu/bnb/tofu_bnb_events.sql:77]: https://github.com/duneanalytics/spellbook/commit/b72d9e294dd7cdccc5330e1d417c319b10f8cb52#r94491148
I mentioned an issue about `nft.trades` on DuneSQL in another place.
Maybe we can patch a quickfix of this line, if this is the cause of the issue.
`, CAST(tff.amount AS DECIMAL(38,0))                 as number_of_items`

Comment (belen-pruvost) [reply models/tofu/bnb/tofu_bnb_events.sql:77]: Indeed - this needs to be explicitly casted to decimal, thank you @Hosuke!

Comment (belen-pruvost) [reply models/tofu/bnb/tofu_bnb_events.sql:77]: @Hosuke has kindly opened a PR to fix this <3 https://github.com/duneanalytics/spellbook/pull/2408

**Pull request merged**